src/type-checker.cc

```diff
@@ -618,15 +617,7 @@ Result TypeChecker::OnEnd() {
   }
 
   const char* desc = s_label_type_name[static_cast<int>(label->label_type)];
-  if (label->label_type == LabelType::Unwind) {
-    // Unwind is unusual in that it always unwinds the control stack at the end,
-    // and therefore the return type of the unwind expressions are not the same
-    // as the block return type.
-    TypeVector empty;
-    result |= OnEnd(label, empty, desc, desc);
-  } else {
-    result |= OnEnd(label, label->result_types, desc, desc);
-  }
+  result |= OnEnd(label, label->result_types, desc, desc);
```

Comment:
I believe it's possible to further simplify here, because the 2nd argument `check_type` (line 592) was added for unwind. The argument can be removed and then `label->result_types` passed in line 596 instead of `check_type`. (see commit 036a632a24679062e7fc891e7743195139bfa0a9 that added this)

Reply:
Thanks! Done.